zcash_client_sqlite/src/wallet.rs

```diff
@@ -779,8 +794,9 @@ pub(crate) fn get_derived_account<P: consensus::Parameters>(
             ":hd_account_index": u32::from(account_index),
```

Comment:
I'm confused: `stmt` uses `:account_id` as a named parameter but the name used here is `:hd_account_index`. How was this working?

Reply:
Hmm. https://docs.rs/rusqlite/latest/rusqlite/trait.Params.html#named-parameters :
> Note: Unbound named parameters will be left to the value they previously were bound with, falling back to NULL for parameters which have never been bound.

Reply:
That's a bug, and we've had that kind of bug before. It would be nice to figure out a way to lint for this.

Reply:
That's a bug, and we've had that kind of bug before. It would be nice to figure out a way to lint for this.

Reply:
I'd settle for a runtime error. Substituting NULL is just terrible. :sob:

Reply:
I guess you could have a macro to generate a struct from the query string, and then you'd pass an instance of that struct rather than `named_params!`. That way the compiler would check whether you'd missed out fields or included undefined fields. Sounds like a lot of effort to get working, though, unless someone has already done something like it.